src/ranged.cpp

```diff
@@ -71,15 +71,98 @@ size_t blood_trail_len( int damage )
+// arcminutes (1/60th of a degree).
+//
+// Increasing dispersion_sigmas will decrease the standard
+// deviation, shots will be "tighter" around the actual
+// target, the overall chance to hit increases.
+//
+// Allowing a hit on the target just because you
+// hit the right tile doesn't work balance-wise as
+// it's too big of a target if you also want to keep
+// miss angles within a reasonable range. So there is
+// a second parameter, occupied_tile_fraction, which says
+// "how wide" the target is. You have to get at least this
+// close with your shot, in tiles (missed_by_tiles)
+// to have a chance to hit. Increasing
+// occupied_tile_fraction increases the chance to hit.
+//
+// You should not set occupied_tile_fraction > 1.0 as
+// that would mean that hitting an adjacent tile is enough
+// to hit the critter, and the code that follows the shot
+// trajectory looking for hits is not capable of handling
+// that.
+//
+// Once you are close enough with the shot for a hit,
+// exactly how close influences the quality of the hit.
+// missed_by represents how good the hit is on a scale
+// of 0 to 1; 0.0 is a perfect hit; 0.99999.. is the
+// worst possible shot that still connects (where you
+// got just barely within occupied_tile_fraction tiles
+// of where you aimed and just clipped the edge of your
+// target)
+//
+// dispersion_sigmas and occupied_tile_fraction can be
+// tuned together to affect the spread of misses without
+// affecting hit or hit quality chances, e.g. if you
+// increase dispersion_sigmas by 10% and also decrease
+// occupied_tile_fraction by 10%, then the cone of misses
+// will get 10% tighter but the overall hit and hit
+// quality chances are unchanged.
+
+// these parameters balance the hit/miss chance:
+const double dispersion_sigmas = 2.4;      // how many standard deviations does "dispersion" represent?
+const double occupied_tile_fraction = 0.5; // how much of the target tile is occupied by the target?
+
 dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg, const tripoint &source,
                                                      const tripoint &target_arg, double dispersion )
 {
     const bool do_animation = get_option<bool>( "ANIMATIONS" );
 
     double range = rl_dist( source, target_arg );
 
-    // an isosceles triangle is formed by the intended and actual targets
-    double missed_by = iso_tangent( range, dispersion );
+    // dispersion is a measure of the dispersion of shots due to the gun + shooter characteristics
+    // i.e. it is independent of any particular shot
+
+    // shot_dispersion is the actual dispersion for this particular shot, i.e.
+    // the error angle between where the shot was aimed and where this one actually went
+    // NB: some cases pass dispersion == 0 for a "never misses" shot e.g. bio_magnet,
+    double shot_dispersion = ( dispersion > 0 ? normal_roll( 0.0, dispersion / dispersion_sigmas ) : 0 );
```

Comment:
`std::max` is not appropriate here because `normal_roll` may return negative values. It's expected that `shot_dispersion` might be negative. This conditional is about giving a different meaning to `dispersion <= 0`, normally (heh) it would be meaningless because the normal distribution is undefined for sigma <= 0